node/src/container_chain_spawner.rs

```diff
@@ -796,7 +810,7 @@ fn handle_update_assignment_state_change(
 /// interrupted before it finished downloading the state, in that case the node will use warp sync.
 /// If it was interrupted during the block history download, the node will use full sync but also
 /// finish the block history download in the background, even if sync mode is set to full sync.
-fn select_sync_mode(
+pub fn select_sync_mode_using_client(
```

Comment:
In this function we only need the `orchestrator_client` to check an edge case for when the container chain is still at block 0. We could also use the relay_chain_interface and get the latest finalized block from there, would that be better for the ws node?

Reply:
we could yes, in the end we are fetching that information from the relay itself

Reply:
Which function in the relay interface should be called? Indeed that would allow to get rid of the generic and not need the orchestrator client.

Reply:
Probably `paras->heads`, but you would need to parse the header to get the block number. There are some utility functions in `pallet_author_noting`, such as `author_from_log`. @girazoki any better ideas? Maybe there is some other storage that only exists after the parachain has included its first block?

Reply:
Can we tackle that in a separate PR?

Reply:
yes let's do it as a separate PR. But it would definitely make it more compatible with solo-chains